be/src/storage/rowset/segment_v2/column_writer.cpp

```diff
@@ -757,15 +755,16 @@ Status ScalarColumnWriter::append(const uint8_t* data, const uint8_t* null_flags
 ////////////////////////////////////////////////////////////////////////////////
 
 ArrayColumnWriter::ArrayColumnWriter(const ColumnWriterOptions& opts, std::unique_ptr<Field> field,
-                                     ScalarColumnWriter* null_writer, ScalarColumnWriter* offset_writer,
+                                     std::unique_ptr<ScalarColumnWriter> null_writer,
+                                     std::unique_ptr<ScalarColumnWriter> offset_writer,
                                      std::unique_ptr<ColumnWriter> element_writer)
         : ColumnWriter(std::move(field), opts.meta->is_nullable()), _element_writer(std::move(element_writer)) {
     if (is_nullable()) {
-        _null_writer.reset(null_writer);
+        _null_writer = std::move(null_writer);
```

Comment:
Please this assignment in the initialization list,  and I think the if condition check is unnecessary.